src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -161,9 +161,6 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
         {
             var modeResult = AttributeMatcher.DetermineMode(context, ModeDetails);
 
-            Debug.Assert(modeResult.FullMatches.Select(match => match.Mode).Distinct().Count() <= 1,
-                $"There should only be one mode match, check the {nameof(ModeDetails)}");
```

Comment:
Getting multiple matches of the same mode is perfectly fine, the code handles it. The important piece is that any attribute that's present that *only* shows up in a partial match is detected and logged (which it is).